src/helpers/api-error-factory.ts

```diff
@@ -15,29 +15,36 @@ import {
   SellingPartnerUnsupportedMediaTypeError,
 } from '../types'
 
-export function apiErrorFactory<T extends ModelError>(error: AxiosError<T>): ExtendableError {
+export function apiErrorFactory<T extends ModelErrorContainer>(
+  error: AxiosError<T>,
+): ExtendableError {
   const { response } = error
   if (response) {
     const { headers, data } = response
+    const modelError = data.errors[0] || {
```

Comment:
@Mykelo Shall we perhaps return an instance of `SellingPartnerUnknownError` here if `data.errors[0]` is `undefined`?

There is no such error class yet, but we can create one.

Reply:
It sounds like a better idea! I implemented it in a new commit and I hope this is what you meant.